connection.go

```diff
@@ -10,21 +12,90 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
+var (
+	ErrNotFound = errors.New("NOT_FOUND")
+)
+
+// extractConnectionNameType extracts the name and connection type from a connection
+// string formatted as "connection://<type>/<name>".
+func extractConnectionNameType(connectionString string) (name string, connectionType string, found bool) {
+	prefix := "connection://"
+
+	if !strings.HasPrefix(connectionString, prefix) {
+		return
+	}
+
+	connectionString = strings.TrimPrefix(connectionString, prefix)
+	parts := strings.SplitN(connectionString, "/", 2)
+	if len(parts) != 2 {
+		return
+	}
+
+	if parts[0] == "" || parts[1] == "" {
+		return
+	}
+
+	return parts[1], parts[0], true
+}
+
+// HydratedConnectionByURL retrieves a connection from the given connection string.
+// The connection string is expected to be of the form: connection://<type>/<name>
+func HydratedConnectionByURL(ctx context.Context, db *gorm.DB, k8sClient kubernetes.Interface, namespace, connectionString string) (*models.Connection, error) {
+	connection, err := FindConnectionByURL(ctx, db, connectionString)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find connection (%s): %w", connectionString, err)
+	}
+
+	if connection == nil {
+		return nil, nil
+	}
+
+	return HydrateConnection(ctx, k8sClient, db, connection, namespace)
+}
+
+// FindConnectionByURL retrieves a connection from the given connection string.
+// The connection string is expected to be of the form: connection://<type>/<name>
+func FindConnectionByURL(ctx context.Context, db *gorm.DB, connectionString string) (*models.Connection, error) {
+	name, connectionType, found := extractConnectionNameType(connectionString)
+	if !found {
+		return nil, nil
+	}
+
+	connection, err := FindConnection(ctx, db, connectionType, name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find connection (type=%s, name=%s): %w", connectionType, name, err)
+	}
+
+	return connection, nil
+}
+
 // FindConnection returns the connection with the given type and name
 func FindConnection(ctx context.Context, db *gorm.DB, connectionType, name string) (*models.Connection, error) {
 	var connection models.Connection
 
 	err := db.Where("type = ? AND name = ?", connectionType, name).First(&connection).Error
-	return &connection, err
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, nil
+		}
+
+		return nil, err
+	}
+
+	return &connection, nil
 }
 
 func GetConnection(ctx context.Context, client kubernetes.Interface, db *gorm.DB, connectionType string, name string, namespace string) (*models.Connection, error) {
 	connection, err := FindConnection(ctx, db, connectionType, name)
 	if err != nil {
 		return nil, err
 	}
-	return HydrateConnection(ctx, client, db, connection, namespace)
 
+	if connection == nil {
+		return nil, ErrNotFound
+	}
```

Comment:
@moshloop we could do this or I could rename the func to `HydratedConnection` like `HydratedConnectionByURL`